Deal with disable option on create

jzaefferer opened the pull request:
Building on top of gh-1599, this fixes missing or redundant handling of the disabled option on create. The only missing widget is tabs. From my comment on the original PR:

> Tabs is not working correctly, for example it ends up with the ui-tabs-disabled class when only setting specific tabs, like disabled: [1, 2]. Due to the array variant I haven't yet figured out what to do here.

jzaefferer commented:
This wasn't really supposed to be a PR on your fork :/

scottgonzalez commented on ui/widgets/accordion.js:
```diff
@@ -175,16 +175,22 @@ return $.widget( "ui.accordion", {
 
+		if ( key === "disabled" ) {
+			this._setOptionDisabled( value );
+		}
```
This already happens in the call to `._super()` above.

scottgonzalez commented on ui/widgets/menu.js:
```diff
@@ -360,12 +355,18 @@ return $.widget( "ui.menu", {
 		if ( key === "disabled" ) {
-			this.element.attr( "aria-disabled", value );
-			this._toggleClass( null, "ui-state-disabled", !!value );
+			this._setOptionDisabled( value );
```
This conditional can be removed since this happens in the call to `._super()`.

scottgonzalez commented on ui/widgets/progressbar.js:
```diff
@@ -118,12 +118,18 @@ return $.widget( "ui.progressbar", {
 		if ( key === "disabled" ) {
-			this.element.attr( "aria-disabled", value );
-			this._toggleClass( null, "ui-state-disabled", !!value );
+			this._setOptionDisabled( value );
```
This can be removed.

scottgonzalez commented on ui/widgets/selectmenu.js:
```diff
@@ -563,17 +559,7 @@ return $.widget( "ui.selectmenu", {
-			} else {
-				this.button.attr( "tabindex", 0 );
-			}
+			this._setOptionDisabled( value );
```
This can be removed.

scottgonzalez commented on ui/widgets/slider.js:
```diff
@@ -431,7 +430,7 @@ return $.widget( "ui.slider", $.ui.mouse, {
 
 		if ( key === "disabled" ) {
-			this._toggleClass( null, "ui-state-disabled", !!value );
+			this._setOptionDisabled( value );
```
This can be removed.

scottgonzalez commented on ui/widgets/spinner.js:
```diff
@@ -429,12 +424,18 @@ $.widget( "ui.spinner", {
-			this._toggleClass( this.uiSpinner, null, "ui-state-disabled", !!value );
-			this.element.prop( "disabled", !!value );
-			this.buttons.button( value ? "disable" : "enable" );
+			this._setOptionDisabled( value );
```
This can be removed.

scottgonzalez commented on ui/widgets/tooltip.js:
```diff
@@ -140,6 +140,8 @@ $.widget( "ui.tooltip", {
 		}
 	},
 
+	_setOptionDisabled: $.noop,
```
Why is this a no-op? We should just have the method call `._disable()` or `._enable()` based on the value. The call to `._disable()` in `._create()` can also be removed then.

scottgonzalez replied:
I'm also not sure why `._setOption()` has the current implementation; removing the `return` shouldn't change the functionality at all, so we should be able to just remove that conditional.

scottgonzalez replied:
Oh, the style changes it's preventing are from the base widget... So, we should still be able to do what I said, the `._setOptionDisabled()` method in tooltip just wouldn't call `._super()`.

scottgonzalez commented:
For tabs, you should be able to just rename `._setupDisabled()`. You can drop the conditional in `._setOptions()` as well.

jzaefferer commented:
Updated. Removed all the redundant calls in `setOption`. Updated tooltip to use `_setOptionDisabled`. 

Updated tabs, renaming to `_setupDisabled` to `_setOptionDisabled`. It still calls `_setOptionDisabled` in `_refresh`, which is also called on create. I added an explicit `_toggleClass` call to only add "ui-tabs-disabled" when all tabs are disabled.
